### PR TITLE
Add patch for accepting non-trec identifiers from WARC files

### LINKED ISSUE #397: Support modern WARC/1.0 inputs in parse_collection
**Describe the bug**
Downloading a WARC file from the common crawl and piping it into `parse_collection` results in an empty forward index.

**To Reproduce**
1. `wget https://commoncrawl.s3.amazonaws.com/crawl-data/CC-NEWS/2018/02/CC-NEWS-20180201022700-00187.warc.gz`
2. `zcat ./CC-NEWS-20180201022700-00187.warc.gz | ./parse_collection -o ./forward -j 5 -f warc --stemmer porter2 --content-parser html`
3. Check file size of the forward index

**Error message**
```
-rw-rw-r-- 1 ubuntu ubuntu    8 Jun  1 01:51 forward
-rw-rw-r-- 1 ubuntu ubuntu   16 Jun  1 01:51 forward.doclex
-rw-rw-r-- 1 ubuntu ubuntu    0 Jun  1 01:51 forward.documents
-rw-rw-r-- 1 ubuntu ubuntu   16 Jun  1 01:51 forward.termlex
-rw-rw-r-- 1 ubuntu ubuntu    0 Jun  1 01:51 forward.terms
-rw-rw-r-- 1 ubuntu ubuntu    0 Jun  1 01:51 forward.urls
```

**Expected behavior**
The warc records should have been extracted/parsed/added to the forward index.

### PR DESCRIPTION
**Pull request by JMMackenzie**
Fixes #397 

Changes proposed in this pull request:

- Uses WARC-Record-ID instead of WARC-TREC-ID where the latter is not available.
- Depends on pending PR here: https://github.com/pisa-engine/warcpp/pull/10
- Will require the submodule update prior to merge

### CODE REVIEW

File: src/parser.cpp [hunk 1]
```diff
@@ -86,8 +86,13 @@ record_parser(std::string const& type, std::istream& is)
                             return std::optional<Document_Record>{};
                         }
                         // TODO(michal): use std::move
-                        return std::make_optional<Document_Record>(
-                            rec.trecid(), rec.content(), rec.url());
+                        if (rec.has_trecid()) {
+                            return std::make_optional<Document_Record>(
+                                rec.trecid(), rec.content(), rec.url());
+                        } else {
```
Comment (amallia) [inline src/parser.cpp]: Could we do `if (rec.has_warcid()) {` instead?
Then have a final `else` case that throws an exception.

**Review by amallia — APPROVED** [review]
Some checks are failing, other than that we can merge it.

File: src/parser.cpp (hunk 1 above)
Comment (JMMackenzie) [reply src/parser.cpp]: No problem. I have just updated this and also updated the `warcpp` submodule too.

**Review by amallia — APPROVED** [review]

**Comment by elshize** [comment]
:rocket:

**Pull request merged**